airflow/providers/delta/sharing/operators/delta_sharing.py

```diff
@@ -0,0 +1,227 @@
+        predicates: Optional[List[str]] = None,
+        save_partitioned: bool = True,
+        save_metadata: bool = True,
+        save_stats: bool = True,
+        overwrite_existing: bool = False,
+        num_parallel_downloads: int = 5,
+        delta_sharing_conn_id: str = 'delta_sharing_default',
+        profile_file: Optional[str] = None,
+        timeout_seconds: int = 180,
+        retry_limit: int = 3,
+        retry_delay: float = 2.0,
+        retry_args: Optional[Dict[Any, Any]] = None,
+        **kwargs,
+    ) -> None:
+        """Creates a new ``DeltaSharingDownloadToLocalOperator``."""
+        super().__init__(**kwargs)
+        self.share = share
+        self.schema = schema
+        self.table = table
+        self.location = location
+        if limit is not None and limit < 0:
+            raise ValueError(f"limit should be greater or equal to 0, got {limit}")
+        self.limit = limit
+        self.predicates = predicates
+        self.save_partitioned = save_partitioned
+        self.save_stats = save_stats
+        self.save_metadata = save_metadata
+        self.overwrite_existing = overwrite_existing
+        if num_parallel_downloads < 1:
+            raise ValueError(
+                "num_parallel_downloads should be greater or equal to 1," f" got {num_parallel_downloads}"
+            )
+        self.num_parallel_downloads = num_parallel_downloads
+
+        self.hook = DeltaSharingHook(
+            delta_sharing_conn_id=delta_sharing_conn_id,
+            retry_args=retry_args,
+            retry_delay=retry_delay,
+            retry_limit=retry_limit,
+            timeout_seconds=timeout_seconds,
+            profile_file=profile_file,
+        )
+
+    def _get_output_file_path(self, metadata: Dict[str, Any], file: Dict[str, Any]) -> str:
+        chunks = urlparse(file['url'])
+        file_name = chunks.path.split('/')[-1]
+        file_parts = [self.location]
+        # add partition parts if table is partitioned
+        partition_values = file.get('partitionValues', {})
+        if self.save_partitioned and len(partition_values) > 0:
+            partitions = metadata.get('partitionColumns', [])
+            for part in partitions:
+                part_value = partition_values.get(part)
+                if part_value is None:
+                    self.log.warning(f"There is no value for partition '{part}'")
+                    part_value = "__null__"
+                file_parts.append(f"{part}={part_value}")
+
+        os.makedirs(os.path.join(*file_parts), exist_ok=True)
+        file_parts.append(file_name)
+        return os.path.join(*file_parts)
+
+    def _download_one(self, metadata: Dict[str, Any], file: Dict[str, Any]):
+        dest_file_path = self._get_output_file_path(metadata, file)
+        file_size = file['size']
+        if os.path.exists(dest_file_path):
+            stat = os.stat(dest_file_path)
+            if file_size == stat.st_size and not self.overwrite_existing:
```

Comment:
Might be safer if we do some sort of MD5 checksum check here and not just the size of the file?